src/main/java/com/tabnine/userSettings/AppSettingsState.kt

```diff
@@ -27,6 +29,11 @@ class AppSettingsState : PersistentStateComponent<AppSettingsState?> {
     var autoImportEnabled: Boolean = true
     var binariesFolderOverride: String = ""
     var cloud2Url: String = ""
+        set(value) {
+            SetCustomRepository(value)
+            RemoveCustomRepository(field)
```

Comment:
i'd create a separate util function `replaceCustomRepository(String old, String new)` - that will call `UpdateSettings.getInstance().getStoredPluginHosts()` only once.
I don't know how this API behaves when called twice after being updated, and even if it works as expected, its more reasonable to perform both changes (delete and add) using the same list obtained by this API.